framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/jobs/JobManager.java

```diff
@@ -26,34 +38,62 @@ public void removeJobListener(IJobListener l) {
 	}
 
 	@Override
-	public void addJob(IJob job) {
+	public IJob scheduleJob(String jobName, IRunnableWithProgress runnable) {
+		SimpleJob job = new SimpleJob(jobName, generateJobId(), runnable);
 		currentJobs.put(job.getId(), job);
-		ArrayList<IJobListener> tmp = new ArrayList<>(listeners);
-		for( IJobListener l : tmp ) {
-			l.jobAdded(job);
-		}
+		fireJobAdded(job);
```

Comment:
shouldn't fire only happen after scheduling was successful? Imagine a listener would ```#cancel(job)``` once it was notified. It couldn't since it would be too early, the job isn't scheduled yet.